source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -12,6 +12,20 @@ namespace Extensions {
 namespace HttpFilters {
 namespace ExtAuthz {
 
+namespace {
+// Asserts that the HTTP method is for a header-only request. For details
+// https://tools.ietf.org/html/rfc7231#section-4.3.
+bool isHeaderOnlyMethod(absl::string_view method) {
+  const static std::vector<std::string>* keys = new std::vector<std::string>(
```

Comment:
`std::unordered_set` or `absl::flat_hash_set` here and find for below

Reply:
:+1: